Fixed TIMESTAMP and TIME truncation

he-is-harry opened the pull request:
- Updated TIMESTAMP and TIME types to truncate milliseconds instead of rounding
- Changed the regex so that fractions of seconds will be read exactly for TIME, TIMESTAMP, and LONGDATE types
    - The fractions are truncated to nearest millisecond for TIME and TIMESTAMP, and nanosecond for LONGDATE
- Added some unit tests for truncating DATETIME types
- Updated integration tests for DFV 4, 5, 6, and 7 types to also be run on DFV 1

Bug: 342310

he-is-harry committed:
Fixed TIMESTAMP and TIME truncation

- Updated TIMESTAMP and TIME types to truncate milliseconds instead of
rounding
- Changed the regex so that fractions of seconds will be read exactly
for TIME, TIMESTAMP, and LONGDATE types
    - The fractions are truncated to nearest millisecond for TIME and
      TIMESTAMP, and nanosecond for LONGDATE
- Added some unit tests for truncating DATETIME types
- Updated integration tests for DFV 4 types to also be run on DFV 1

he-is-harry committed:
Changed DFV5 and DFV7 integration tests to also run on DFV 1

- Updated the spatial types and boolean integration tests to also
be run on DFV 1 (BINTEXT DFV6 integration tests were updated in
the previous commit)
- Wrote a comment about a BOOLEAN behaviour change in DFV 7, where
the empty string '' will result in null in DFV 7 but in DFV 1 it
results in false

IanMcCurdy reviewed (changes requested):
Change looks good. After you open a bugzilla for the truncation/rounding issue just add the bug number into the commit description

he-is-harry committed:
Removed comment about empty string boolean input

- Removed the comment about the behaviour change in DFV 7 which
affects empty string inputs for BOOLEAN.
    - The behaviour will be clear in the test change for BOOLEAN
    in a separate pull request

Bug: 342310

he-is-harry committed:
Merge branch 'master' into truncateTimestamp

- Updated the call of setUpTableRemoteDB in the FIXED integration
tests to match with new consolidated setUpTable introduced in
fix TIMESTAMP and TIME commit to simplify the data type integration
tests
